http_api.go

```diff
@@ -0,0 +1,194 @@
+package worker
+
+import (
+	"crypto/subtle"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// APIHandler handles requests to worker's HTTP API.
+type APIHandler struct {
+	i *CLI
+}
```

Comment:
There are some other related comments, but I think I'm leaning toward this struct having access/construction more like:

``` go
type APIHandler struct {
  processorPool *ProcessorPool
  auth string
  info func() *workerInfo
  cancel func()
}
```

Reply:
Okay, I can get behind that. The last two function strike me as a little strange to bring in this way, but I'm not sure. Maybe we can reconsider later if we end up needing more of `CLI`'s methods in the future.